core/app/models/workarea/order/item.rb

```diff
@@ -166,5 +166,22 @@ def customizations_eql?(test)
         test.blank? && customizations.blank?
       end
     end
+
+    # Determine whether the additional fields on this item are
+    # equivalent to the passed-in Hash of parameters. Uses the
+    # `distinct_order_item_attributes` configuration setting to filter
+    # down which parameters it will be checking equivalence on.
+    #
+    # @param [Hash] params - Updated order item parameters.
+    # @return [Boolean] whether this item's attributes is equal to the
+    #                   passed-in parameters.
+    def attributes_eql?(params = {})
+      equivs = Workarea.config.distinct_order_item_attributes.map(&:to_sym)
+      attrs = params&.symbolize_keys&.slice(*equivs)
+
+      equivs.blank? || params.blank? || attrs.all? do |attribute, value|
```

Comment:
I think this should be false if the configuration is blank - by setting those blank you're saying "there are no attributes that make 2 order items equivalent".

Also `all?` will return `true` on an empty enumerable so this logic needs some clean up.

Reply:
> by setting those blank you're saying "there are no attributes that make 2 order items equivalent".

That's what preserves the original functionality, because before this code was introduced, there _were_ no attributes that made two order items equivalent, other than customizations, which if `customizations_eql?` returns true this method isn't even called.